src/components/generic/Table/Table.tsx

```diff
@@ -97,7 +101,10 @@ export default class Table extends GenericComponent<ITableProps, ITableState> {
                         <span className="primary">{value[col.field]}</span>
                         <span className="secondary">{value[col.secondaryField]}</span>
                       </div>
-                    ) : value[col.field]
```

Comment:
I believe a switch statement would be cleaner, and perhaps to separate the switch to a new smaller function or smaller render chunks